modules/openapi-generator/src/main/resources/dart-dio/api.mustache

```diff
@@ -88,12 +88,12 @@ class {{classname}} {
 
         {{#isListContainer}}
                 final FullType type = const FullType(BuiltList, const [const FullType({{returnBaseType}})]);
-                BuiltList<{{returnBaseType}}> dataList = _serializers.deserialize(jsonDecode(response.data), specifiedType: type);
+                BuiltList<{{returnBaseType}}> dataList = _serializers.deserialize(response.data is String ? jsonDecode(response.data) : response.data, specifiedType: type);
```

Comment:
After this change, you cannot have a 'text/plain' response, as it tries to apply jsonDecode on it.

:(

Reply:
With  'text/plain'  response you shouldn't arrived here as you don't need to deserialize at all

Reply:
Indeed, but it was trying to deserialize a text/plain response.